admin/spam/views.py

```diff
@@ -1,9 +1,42 @@
-from django.http import HttpResponse
+from django.shortcuts import render, render_to_response
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+from django.contrib.auth.decorators import login_required
 
+from .serializers import serialize_comments, retrieve_comment
 
+
+@login_required
 def spam_list(request):
-    return HttpResponse('This is a list of spam')
+    comment_list = serialize_comments()
```

Comment:
You are loading all comments that have a report here, even if the client only requests a single page. You should only need to load the records that the clients asks for. You should be able to use the `limit` and `offset` functionality of modm to achieve this.